pkgs/by-name/af/affine/package.nix

```diff
@@ -105,142 +78,157 @@ stdenv.mkDerivation (
+        mkdir -p $out
+        yarn config set enableTelemetry false
+        yarn config set cacheFolder $out
+        yarn config set enableGlobalCache false
+        yarn config set supportedArchitectures --json "${supportedArchitectures}"
 
-        # second build
-        yarn config set nmMode classic
-        yarn config set nmHoistingLimits workspaces
-        find . -name 'node_modules' -type d -prune -exec rm -rf '{}' +
-        yarn workspaces focus @affine/electron @affine/monorepo
-        BUILD_TYPE=${buildType} SKIP_WEB_BUILD=1 SKIP_BUNDLE=1 HOIST_NODE_MODULES=1 yarn workspace @affine/electron make
+        yarn install --immutable --mode=skip-build
 
         runHook postBuild
       '';
-      installPhase =
-        let
-          inherit (finalAttrs) binName productName;
-        in
-        if hostPlatform.isDarwin then
-          ''
-            runHook preInstall
-
-            mkdir -p $out/Applications
-            mv packages/frontend/apps/electron/out/${buildType}/${productName}-darwin-${nodeArch}/${productName}.app $out/Applications
-
-            runHook postInstall
-          ''
-        else
-          ''
-            runHook preInstall
-
-            mkdir --parents $out/lib/${binName}/
-            mv packages/frontend/apps/electron/out/${buildType}/${productName}-linux-${nodeArch}/{resources,LICENSE*} $out/lib/${binName}/
-            install -Dm644 packages/frontend/apps/electron/resources/icons/icon_${buildType}_64x64.png $out/share/icons/hicolor/64x64/apps/${binName}.png
-
-            makeWrapper "${electron}/bin/electron" $out/bin/${binName} \
-              --inherit-argv0 \
-              --add-flags $out/lib/${binName}/resources/app.asar \
-              --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations --enable-wayland-ime=true}}" \
-              --add-flags ${lib.escapeShellArg commandLineArgs}
-
-            runHook postInstall
-          '';
-    }
-    // (lib.optionalAttrs hostPlatform.isLinux {
-      desktopItems =
-        let
-          inherit (finalAttrs) binName productName;
-        in
-        [
-          (makeDesktopItem {
-            name = binName;
-            desktopName = productName;
-            comment = "AFFiNE Desktop App";
-            exec = "${binName} %U";
-            terminal = false;
-            icon = binName;
-            startupWMClass = binName;
-            categories = [ "Utility" ];
-            mimeTypes = [ "x-scheme-handler/${binName}" ];
-          })
-        ];
+    dontInstall = true;
+    outputHashMode = "recursive";
+    outputHash = "sha256-E9l5zjOOfyDBzYJOU94VrRvt7Hi4XkRTDav9bVlXvlQ=";
+  };
+  nativeBuildInputs =
+    [
+      nodejs
+      yarn
+      cargo
+      rustc
+      findutils
+      zip
+      jq
+      rsync
+      writableTmpDirAsHomeHook
+    ]
+    ++ lib.optionals hostPlatform.isLinux [
+      copyDesktopItems
+      makeWrapper
+    ];
+
+  env.CI = "1";
```

Comment:
Optional: Would be nice to have a comment about this.

Reply:
Ok, let me add it